### ISSUE TITLE
Jupyter Notebook no longer automatically opens in browser

### ISSUE THREAD

**Issue by etadobson** [jupyter/notebook#5361]
I recently [created a StackOverflow post to address my issue](https://stackoverflow.com/questions/60959557/jupyter-notebook-no-longer-automatically-opens-in-browser). However, no one has responded.  I will repeat my issue here. 

When I go to run jupyter notebooks via terminal... calling

```jupyter notebook```

I see:
```
[I 15:33:26.341 NotebookApp] JupyterLab extension loaded from /Users/ellenarena/opt/anaconda3/envs/ellenDL/lib/python3.8/site-packages/jupyterlab
[I 15:33:26.341 NotebookApp] JupyterLab application directory is /Users/ellenarena/opt/anaconda3/envs/ellenDL/share/jupyter/lab
[I 15:33:26.344 NotebookApp] Serving notebooks from local directory: /Users/ellenarena
[I 15:33:26.344 NotebookApp] The Jupyter Notebook is running at:
[I 15:33:26.344 NotebookApp] http://localhost:8888/?token=2e39eea78241a1f47a1071cad5210ba3531007b7b7f73e0a
[I 15:33:26.344 NotebookApp]  or http://127.0.0.1:8888/?token=2e39eea78241a1f47a1071cad5210ba3531007b7b7f73e0a
[I 15:33:26.344 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 15:33:26.356 NotebookApp] 

    To access the notebook, open this file in a browser:
        file:///Users/ellenarena/Library/Jupyter/runtime/nbserver-24533-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=2e39eea78241a1f47a1071cad5210ba3531007b7b7f73e0a
     or http://127.0.0.1:8888/?token=2e39eea78241a1f47a1071cad5210ba3531007b7b7f73e0a
5:15: syntax error: A “"” can’t go after this identifier. (-2740)
```

However, no browser window automatically opens... I have to copy/paste the URL. The notebooks work/run... but this is a bit of a pain having to stop and copy/paste to open things.

It's not a problem with my browser - was working fine before. I did do a recent fresh install of Anaconda and the location of anaconda3/ folder has moved... could that be a reason why?

Also in case it's need - here is the version of Jupyter I'm running:
```
jupyter core     : 4.6.3
jupyter-notebook : 6.0.3
qtconsole        : 4.7.2
ipython          : 7.13.0
ipykernel        : 5.2.0
jupyter client   : 6.1.2
jupyter lab      : 2.0.1
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.4
traitlets        : 4.3.3
```
Any help would be much appreciated.

Running on a Mac 10.14.6 with Python 3.8.2 and conda 4.8.3.

**Comment by jpcastil** [comment]
I could try to replicate this issue if you detail the steps

**Comment by etadobson** [comment]
I'm not sure what exact steps I can provide... only that when I go to my terminal and try to open jupyter notebooks by calling ```jupyter notebook```, the interface does not directly open in my browser (as it used to...).   I now have to copy/paste link provided into the browser (from what is displayed in the terminal window after the ```jupyter notebook``` call I type - see the text above that I shared).  Is there more specific/better information I can provide?  I really do appreciate your help...

**Comment by etadobson** [comment]
It has resolved itself somehow and now open automatically.  Thank you.

**Comment by paulsanjib** [comment]
I also faced the same issue and did the following to resolve it. You can try 

Add this line to your jupyter_notebook_config.py file:
c.NotebookApp.use_redirect_file = False

The config file should be found in ~/.jupyter/jupyter_notebook_config.py. If missing can be generated (as you did) with the command: jupyter notebook --generate-config

**Comment by pkrezel** [comment]
Thank you Paulsanjib, it works!

**Comment by ekaharo** [comment]
Worked very well...thanks for sharing

**Comment by ChaoB16** [comment]
a) Hi everybody I have the same problem (can't open automatically), plus when i wan't to delete a file on the notebook it doesn't let me. 

b) I'have tried the commands of @paulsanjib but they don't work with me. I am just beginning to learn python so maybe i did a mistake somehow. Running on macOS 10.13.6. 

Any help would be appreciated thank you.

a) 
(base) macbook-pro-de-adrien-3:~ adrienchao$ jupyter notebook
[I 09:55:50.800 NotebookApp] JupyterLab extension loaded from /opt/anaconda3/lib/python3.8/site-packages/jupyterlab
[I 09:55:50.800 NotebookApp] JupyterLab application directory is /opt/anaconda3/share/jupyter/lab
[I 09:55:50.806 NotebookApp] Serving notebooks from local directory: /Users/adrienchao
[I 09:55:50.806 NotebookApp] Jupyter Notebook 6.1.4 is running at:
[I 09:55:50.806 NotebookApp] http://localhost:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
[I 09:55:50.806 NotebookApp]  or http://127.0.0.1:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
[I 09:55:50.806 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 09:55:50.812 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///Users/adrienchao/Library/Jupyter/runtime/nbserver-48415-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
     or http://127.0.0.1:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
5:15: syntax error: « « " » » ne peut pas se trouver après « identificateur ». (-2740)

b) 
(base) macbook-pro-de-adrien-3:~ adrienchao$ jupyter notebook --generate-config
Overwrite /Users/adrienferrin/.jupyter/jupyter_notebook_config.py with default config? [y/N]y
Writing default config to: /Users/adrienchao/.jupyter/jupyter_notebook_config.py
(base) macbook-pro-de-adrien-3:~ adrienchao$ c.NotebookApp.use_redirect_file = False
-bash: c.NotebookApp.use_redirect_file: command not found
(base) macbook-pro-de-adrien-3:~ adrienchao$

**Comment by ChaoB16** [comment]
a) Hi everybody I have the same problem (can't open automatically), plus when i wan't to delete a file on the notebook it doesn't let me.

b) I'have tried the commands of @paulsanjib but they don't work with me. I am just beginning to learn python so maybe i did a mistake somehow. Running on macOS 10.13.6.

Any help would be appreciated thank you.

a)
(base) macbook-pro-de-adrien-3:~ adrienchao$ jupyter notebook
[I 09:55:50.800 NotebookApp] JupyterLab extension loaded from /opt/anaconda3/lib/python3.8/site-packages/jupyterlab
[I 09:55:50.800 NotebookApp] JupyterLab application directory is /opt/anaconda3/share/jupyter/lab
[I 09:55:50.806 NotebookApp] Serving notebooks from local directory: /Users/adrienchao
[I 09:55:50.806 NotebookApp] Jupyter Notebook 6.1.4 is running at:
[I 09:55:50.806 NotebookApp] http://localhost:8888/?token=fb14611541c1454443d3859f(...)
[I 09:55:50.806 NotebookApp] or http://127.0.0.1:8888/?token=fb14611541c1454443d38(...)
[I 09:55:50.806 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 09:55:50.812 NotebookApp]

To access the notebook, open this file in a browser:
    file:///Users/adrienchao/Library/Jupyter/runtime/nbserver-48415-open.html
Or copy and paste one of these URLs:
    http://localhost:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
 or http://127.0.0.1:8888/?token=fb14611541c1454443d3859f8a8c91c24715f42a3765b11e
5:15: syntax error: « « " » » ne peut pas se trouver après « identificateur ». (-2740)

b)
(base) macbook-pro-de-adrien-3:~ adrienchao$ jupyter notebook --generate-config
Overwrite /Users/adrienchao/.jupyter/jupyter_notebook_config.py with default config? [y/N]y
Writing default config to: /Users/adrienchao/.jupyter/jupyter_notebook_config.py
(base) macbook-pro-de-adrien-3:~ adrienchao$ c.NotebookApp.use_redirect_file = False
-bash: c.NotebookApp.use_redirect_file: command not found
(base) macbook-pro-de-adrien-3:~ adrienchao$

**Comment by kevin-bates** [comment]
Hi @ChaoB16.

I can't tell what the syntax error is in section a).  Google translate indicates something about an identifier not found issue.  Are you displaying the actual console output?  Is this occurring when attempting to delete a file (since you indicate that's an issue as well) or just after starting the notebook server?

Regarding section b).  `c.NotebookApp.use_redirect_file = False` is the syntax used in the notebook configuration file (`jupyter_notebook_config.py`) to set the configurable attribute `use_redirect_file`.  This option can also be set from the command line when launching the notebook server:
```
jupyter notebook --NotebookApp.use_redirect_file=False
```
Note that there's no `c.` prefix when using the command line option.  I would also recommend using `--debug` to enable DEBUG logging until you've gotten past these issues.

**Comment by ChaoB16** [comment]
Hi @kevin-bates 

Yes you're right on the traduction, and yes I am displaying the output of the console. The problem, was that when I launched the notebook it didn't popup automatically but instead I had to copy and paste the link. I could get along with this, but when I deleted a file on the notebook it didn't let me and i had an error message. But funny thing, somehow it got fixed by himself (because since I asked for help I didn't get the time to touch it) and now it's working fine haha. Quite disconcerting. 

Thank you very much @kevin-bates for the help, I really appreciate I will keep your instructions in the case it's going nuts again.

Have a good day. 
Adrien
